docs/sections/contributor_guide/code_quality.rst

```diff
@@ -1,5 +1,36 @@
-************
 Code Quality
-************
+============
 
-TBD
+.. include:: links.rst
+
+In an active development shell, the following ``make`` targets are available and act on all ``.py`` files under ``src/``:
+
+.. list-table::
+   :widths: 15 85
+   :header-rows: 1
+
+   * - Command
+     - Description
+   * - ``make format``
+     - Formats code with `black`_, imports with `isort`_, docstrings with `docformatter`_, and ``.jsonschema`` documents with `jq`_
+   * - ``make lint``
+     - Lints code with `pylint`_
+   * - ``make typecheck``
+     - Typechecks code with `mypy`_
+   * - ``make unittest``
+     - Runs unit tests and reports coverage with `pytest`_ and `coverage`_
+   * - ``make test``
+     - Equivalent to ``make lint && make typecheck && make unittest``, plus checks defined CLI scripts.
+
+Configuration for these tools is provided by the file ``src/pyproject.toml``.
+
+The order of the targets above is intentional:
+
+   * ``make format`` will complain about certain kinds of syntax errors that would cause all the remaining code-quality tools to fail (and may change line numbers reported by other tools, if it ran after them).
+   * ``make lint`` provides a good first check for obvious errors and anti-patterns in the code.
+   * ``make typecheck`` offers a more nuanced look at interfaces between functions, methods, etc. and may spot issues missed by the linter.
+   * ``make unittest`` provides higher-level semantic-correctness checks once code syntax and typing is deemed correct.
+
+All the above tests are executed by the CI system when code is merged to specific git branches, and again when a conda package is built for release. To ensure that these processes succeed, be sure to run the tests and format the code in the development shell before opening a pull request, and throughout the PR's lifecycle if changes are made and pushed. Note that ``make format`` requires explicit manual execution by the developer: It is not run automatically by ``make test``, to avoid reformatting under-development code in a way that might surprise the developer. A useful development idiom is to periodically run ``make format && make test`` to perform a full code-quality sweep through the code. The CI system is configured to rejected unformatted code, so formatting via ``make format`` is first-class code-quality citizen.
```

Comment:
I'm unsure about this punctuation, specifically the colon and the comma after ``make test``...
```
Note that ``make format`` requires explicit manual execution by the developer: It is not run automatically by ``make test``, to avoid reformatting under-development code in a way that might surprise the developer. 
```

(Apparently Grammarly is fine with the colon, but is suggesting the removal of the comma)

I might also suggest changing it to read "to avoid reformatting code that is under development"

Perhaps change the last bit to say "is a first-class code-quality citizen"?

Reply:
Removing the comma would confuse the meaning, which Grammarly might not have understand. For example, if I wrote
> I did not shout, to avoid waking the baby.

the phrase after the comma is clearly a justification for the preceding phrase, while if I wrote

> I did not shout to avoid waking the baby.

one might ask "Well, why _did_ you shout, then?" It's similar here. I could rewrite the first sentence as
> To avoid waking the baby, I did not shout.

so maybe a different construction here would help. I think it's down to emphasis: My first draft here states the _fact_ ASAP, then provides the justification. The other way around makes the reader wait longer for the crucial point: That the formatters do not run automatically.

> to avoid reformatting code that is under development

I'm willing to change it, though I consider that kind phrasing wordy and weak. For example, "blue-green sea" is concise and punchy compared to "sea that is blue-green". The old network TV announcement said "We now return to our regularly scheduled program", not "We now return to our program that is regularly scheduled." The suggested edit is definitely less weird than those!

(I love nitpicking English -- and looking for the best turn of phrase -- as much as I do code, which is a lot... when I have time... which I apparently do.)

> is first-class code-quality citizen

is straight-up typo! Will fix.